src/components/Summary.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState } from 'react';
+import { css } from 'emotion';
+import { textSans, headline, body } from '@guardian/src-foundations/typography';
+import {
+    news,
+    neutral,
+    opinion,
+    culture,
+    sport,
+    lifestyle,
+    labs,
+} from '@guardian/src-foundations/palette/';
+import { SvgMinus, SvgPlus } from '@guardian/src-icons';
+
+/// SUMMARY ELEMENT
+let linkColourStyleHover = '';
+let titleColour = '';
+
+const titleStyling = css`
+    ${headline.xxxsmall({
+        fontWeight: 'medium',
+    })};
+    margin: 0;
+    line-height: 22px;
+`;
+
+const showHideStyling = css`
+    background: ${neutral[7]};
+    color: ${neutral[100]};
+    height: 2rem;
+    position: absolute;
+    bottom: 0;
+    transform: translate(0, 50%);
+    padding: 0 15px 0 7px;
+    border-radius: 100em;
+    cursor: pointer;
+    display: inline-flex;
+    align-items: center;
+    border: 0;
+    margin: 0;
+`;
+
+const plusStyling = css`
+    margin-right: 12px;
+    margin-bottom: 6px;
+    width: 33px;
+    fill: white;
+    height: 28px;
+`;
+
+const minusStyling = css`
+    margin-right: 14px;
+    margin-bottom: 6px;
+    width: 30px;
+    fill: white;
+    height: 25px;
+    padding-left: 4px;
+`;
+
+const iconSpacing = css`
+    display: inline-flex;
+    align-items: center;
+    ${textSans.small()};
+`;
+
+function SetHighlightColour(pillar: string) {
+    switch (pillar) {
+        case 'opinion':
+            linkColourStyleHover = opinion[400];
+            titleColour = opinion[400];
+            break;
+        case 'sport':
+            linkColourStyleHover = sport[400];
+            titleColour = sport[400];
+            break;
+        case 'culture':
+            linkColourStyleHover = culture[400];
+            titleColour = culture[400];
+            break;
+        case 'lifestyle':
+            linkColourStyleHover = lifestyle[400];
+            titleColour = lifestyle[400];
+            break;
+        case 'labs':
+            linkColourStyleHover = labs[400];
+            titleColour = labs[400];
+            break;
+        case 'news':
+            linkColourStyleHover = news[400];
+            titleColour = news[400];
+            break;
+    }
+}
```

Comment:
Setting `let` variables can make it difficult to follow the code, so using a more functional approach is favoured where we `return` the colour (especially as they're all the same):

```suggestion
function getPillarHighlight(pillar: Pillar) { 
    switch (pillar) {
        case 'opinion':
            return opinion[400];
        case 'sport':
			return sport[400];
        case 'culture':
            return culture[400];
        case 'lifestyle':
            return lifestyle[400];
        case 'labs':
			return labs[400];
        case 'news':
        default:
			return news[400];
    }
}
```

and use it in the css like: 

```
background: ${getPillarHighlight(pillar)};
```

Reply:
I think you probably want to pull it into a `lib` function as well as you've got a few places you have this pattern that you'll want to remove